loudcloud-dev/ee_calculator
# feat: leave approvals

## Changelog

### Features

#### Model
- added field to `Leave` model
    - `processed_by` - associated with admin user who approves/rejects the leave

#### Controller
- added custom Active Admin actions
    - `approve` - sets leave status to "approved", and "processed_by" to current admin user
    - `reject` - sets leave status to "rejected", and "processed_by" to current admin user

#### View
- added leaves to Active Admin
    - added approve and reject actions on index view and single view
        - should show up only on pending leaves
    - added color badges to status
 
## How to check
- log in to an employee account and file leaves
- log in to an admin account and go to leaves module
- verify that all filters are working correctly
- verify that "Approve" and "Reject" buttons appear on index view and single view for pending leaves
- verify that the buttons properly update `status` and `processed_by` fields
- verify that approved/rejected leaves won't have the buttons anymore
